check_format.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const glob = require('glob');
+
+const CLANG_FORMAT = 'node_modules/.bin/clang-format';
+
+/**
+ * Checks that a path is formatted, fixing it if fix is true.
+ * @return false if the path was not formatted and we didn't fix it.
+ */
+function checkFile(path, fix) {
+  const source = fs.readFileSync(path, 'utf8');
+  const formatted = childProcess.execFileSync(CLANG_FORMAT, [path], {encoding: 'utf8'});
+  if (source !== formatted) {
+    if (!fix) return false;
+    fs.writeFileSync(path, formatted);
+  }
+  return true;
+}
+
+function main(args) {
+  const fix = args[0] === '--fix';
+  const sourceFiles = glob.sync('{*.js,src/**/*.[jt]s,test/**/*.[jt]s}');
+  for (const path of sourceFiles) {
+    if (!checkFile(path, fix)) {
+      console.error(`${path} not formatted; run 'node check_format.js --fix'`);
```

Comment:
The old version of this only printed one command if you needed to fix it.  Could you accumulated failing paths and print once?

Reply:
The 'return' statement right after means this only prints once.  Did you mean you want the error message to list out which files are wrong?

Reply:
I wasn't seeing the return, so I guess this is ok.  I just didn't want users to see 5 'to format, run COMMAND' messages in a row.